admin/class-yoast-alerts.php

```diff
@@ -0,0 +1,245 @@
+	}
+
+	/**
+	 * Enqueue assets
+	 */
+	public function enqueue_assets() {
+
+		$asset_manager = new WPSEO_Admin_Asset_Manager();
+		$asset_manager->enqueue_style( 'alerts' );
+	}
+
+	/**
+	 * Handle ajax request to dismiss an alert
+	 */
+	public function ajax_dismiss_alert() {
+
+		$notification = $this->get_notification_from_ajax_request();
+		if ( $notification ) {
+			$notification_center = Yoast_Notification_Center::get();
+			$notification_center->maybe_dismiss_notification( $notification );
+
+			$this->output_ajax_response( $notification->get_type() );
+		}
+		die();
+	}
+
+	/**
+	 * Handle ajax request to restore an alert
+	 */
+	public function ajax_restore_alert() {
+
+		$notification = $this->get_notification_from_ajax_request();
+		if ( $notification ) {
+			delete_user_meta( get_current_user_id(), $notification->get_dismissal_key() );
+
+			$this->output_ajax_response( $notification->get_type() );
+		}
+
+		die();
+	}
+
+	/**
+	 * Create AJAX response data
+	 *
+	 * @param string $type Alert type.
+	 */
+	private function output_ajax_response( $type ) {
+
+		$html = $this->get_view_html( $type );
+		echo json_encode(
+			array(
+				'html'  => $html,
+				'total' => count( self::$active_errors ) + count( self::$active_warnings ),
+			)
+		);
+	}
+
+	/**
+	 * Get the HTML to return in the AJAX request
+	 *
+	 * @param string $type Alert type.
+	 *
+	 * @return bool|string
+	 */
+	private function get_view_html( $type ) {
+
+		switch ( $type ) {
+			case 'error':
+			case 'warning':
+				$view = $type . 's';
+				break;
+			default:
+				return false;
+		}
+
+		// Re-collect alerts.
+		self::collect_alerts();
+
+		/** @noinspection PhpUnusedLocalVariableInspection */
+		$alerts_data = self::get_template_variables();
+
+		ob_start();
+		include WPSEO_PATH . 'admin/views/partial-alerts-' . $view . '.php';
+		$html = ob_get_clean();
+
+		return $html;
+	}
+
+	/**
+	 * Extract the Yoast Notification from the AJAX request
+	 *
+	 * @return null|Yoast_Notification
+	 */
+	private function get_notification_from_ajax_request() {
+
+		$notification_center = Yoast_Notification_Center::get();
+		$notification_id     = filter_input( INPUT_POST, 'notification' );
+
+		return $notification_center->get_notification_by_id( $notification_id );
+	}
+
+	/**
+	 * Show the alerts overview page
+	 */
+	public static function show_overview_page() {
+
+		/** @noinspection PhpUnusedLocalVariableInspection */
+		$alerts_data = self::get_template_variables();
+
+		include WPSEO_PATH . 'admin/views/alerts-dashboard.php';
+	}
+
+	/**
+	 * Collect the alerts and group them together
+	 */
+	public static function collect_alerts() {
+
+		$notification_center = Yoast_Notification_Center::get();
+
+		$notifications            = $notification_center->get_notifications();
+		self::$notification_count = count( $notifications );
+
+		self::$errors           = array_filter( $notifications, array( __CLASS__, 'filter_error_alerts' ) );
```

Comment:
I see two types of notifications. Maybe it's nice to have a notification container class with two instances; one for errors and one for warnings.